issuetracker/src/main/java/org/openjdk/skara/issuetracker/jira/JiraIssue.java

```diff
@@ -291,9 +298,13 @@ public void setLabels(List<String> labels) {
 
     @Override
     public List<Label> labels() {
-        return json.get("fields").get("labels").stream()
-                   .map(s -> new Label(s.asString()))
-                   .collect(Collectors.toList());
+        if(labels == null){
+            json = request.get("").execute();
```

Comment:
I don't think it's a good idea to have `JiraIssue::labels` sometimes update the whole json object. Just fetch it to a local variable and keep the `json` field final.